integrations/mailer/mailer.py

```diff
@@ -256,6 +256,13 @@ def send_email(self, alert):
                                      ' adding for this rule only')
                             del contacts[:]
                             contacts.extend(new_contacts)
+        
+        # Don't loose time (and try to send an email) if there is no contact...
+        for contact in contacts:
+            if contact == '':
+                contacts.remove(contact)
+        if len(contacts) == 0:
```

Comment:
I think you'll find you can rewrite this conditional as...
```
if not contacts:
    return
```

Reply:
Not sure that this will handle the case "contact list is not empty, but with only one item which is empty string". I had the case when not defining the mail_to parameter at all, contacts was then equal to [''], I didn't investigated more though, I'll have a look tomorrow, perhaps that comes from something else that could be corrected to.

Reply:
I'm only suggesting this as replacement for lines 264-265. Not the whole thing.

Reply:
Yes, sure, but I am not confortable with the rest of the patch (the part removing the '' string).
I ended up analysing the options building code to fix the problem before (at config parameters handling).
The problem was that the rawconfigparser returned an empty string if the parameter was an empty list. Then ''.split(',') would return [''] (this is advertised in str.split manual).

The new proposed patch solves that.